pkg/activator/net/revision_backends.go

```diff
@@ -259,28 +259,42 @@ func (rw *revisionWatcher) checkDests(dests []string) {
 	}
 }
 
-func (rw *revisionWatcher) runWithTickCh(tickCh <-chan time.Time) {
+func (rw *revisionWatcher) run(probeFrequency time.Duration) {
 	var dests []string
+	var timer *time.Timer
+
 	for {
-		select {
-		case <-rw.doneCh:
-			return
-		case x, ok := <-rw.destsChan:
-			if !ok {
-				return
+		if func() bool {
+			var tickCh <-chan time.Time
+
+			if len(dests) != 0 && !rw.clusterIPHealthy {
+				if timer == nil {
+					timer = time.NewTimer(probeFrequency)
+				} else {
+					timer.Reset(probeFrequency)
+				}
+
+				defer timer.Stop()
```

Comment:
I'd really rather avoid the defer and func() wrapper.  I think the intent was to create the ticker outside of the loop and then each iteration either Stop or Reset it.

Reply:
done